.github/workflows/ci.yml

```diff
@@ -1,78 +1,129 @@
-name: CI
+name: Kelvin CI
 on:
   pull_request:
+      paths-ignore:
+        - 'deployment_service/**'
+        - '.github/workflows/deployment_service-ci.yml'
   merge_group:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
+  cancel-in-progress: ${{ github.event_name != 'merge_group' }}
+
+env:
+  # Configure a constant location for the uv cache
+  UV_CACHE_DIR: /tmp/.uv-cache
+
 jobs:
   test-backend:
     runs-on: ubuntu-latest
+
     steps:
       - name: Checkout sources
-        uses: actions/checkout@v4
-      - name: Install Python
+        uses: actions/checkout@v5
+
+      - name: Setup Python
+        id: setup_python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.12"
+          python-version-file: "pyproject.toml"
+
+      - name: Install the latest version of uv
+        uses: astral-sh/setup-uv@v7
+        with:
+          enable-cache: "auto"
+          version: "0.9.9"
+
+      - name: Install OS dependencies
+        run: |
+          sudo apt-get install -y libgraphviz-dev build-essential
+
+      - name: Cache venv
```

Comment:
It takes ~25s to install dependencies. I'm pretty sure that caching them isn't worth it. Let's remove this.